src/Internal.AspNetCore.BuildTools.Tasks/SetEnvironmentVariable.cs

```diff
@@ -0,0 +1,32 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using Microsoft.Build.Utilities;
+using Microsoft.Build.Framework;
+
+namespace Microsoft.AspNetCore.BuildTools
+{
+    public class SetEnvironmentVariable : Task
+    {
+        [Required]
+        public string Variable { get; set; }
+
+        [Required]
+        public string Value { get; set; }
+
+        public override bool Execute()
+        {
+            if (string.IsNullOrEmpty(Variable))
+            {
+                Log.LogError($"{nameof(Variable)} cannot be null or an empty string");
+                return false;
+            }
+
+            Log.LogMessage("Setting env '{0}' = '{1}'", Variable, Value);
```

Comment:
Let's expand this a little so it looks like a normal build message: "Setting Environment Variable '{0}' to '{1}'" (and while you're at it, using an interpolated string)

(And yes, I'm aware this is code you copied from me 😝)